net/sched/act_skbedit.c

```diff
@@ -39,7 +42,30 @@ static int tcf_skbedit(struct sk_buff *skb, const struct tc_action *a,
 	tcf_lastuse_update(&d->tcf_tm);
 	bstats_update(&d->tcf_bstats, skb);
 
-	if (d->flags & SKBEDIT_F_PRIORITY)
+	if (d->flags & SKBEDIT_F_INHERITDSFIELD) {
+		int wlen = skb_network_offset(skb);
+
+		switch (tc_skb_protocol(skb)) {
+		case htons(ETH_P_IP):
+			wlen += sizeof(struct iphdr);
+			if (!pskb_may_pull(skb, wlen))
+				goto err;
+			skb->priority = ipv4_get_dsfield(ip_hdr(skb)) >> 2;
+			break;
+
+		case htons(ETH_P_IPV6):
+			wlen += sizeof(struct ipv6hdr);
+			if (!pskb_may_pull(skb, wlen))
+				goto err;
+			skb->priority = ipv6_get_dsfield(ipv6_hdr(skb)) >> 2;
+			break;
+
+		default:
+			if (d->flags & SKBEDIT_F_PRIORITY)
+				skb->priority = d->priority;
+			break;
+		}
+	} else if (d->flags & SKBEDIT_F_PRIORITY)
 		skb->priority = d->priority;
```

Comment:
1. Encircle the `else` clause with braces (i.e. `{` and `}`);
2. Add the label `default_priority:` to the very beginning of the `else` clause, that is, before `if (d->flags & SKBEDIT_F_PRIORITY)`;
3. Replace the body of the default clause of the `switch` above with `goto default_priority;`.